HDCP Daemon integrating with IAS compositor

saik-intel opened the pull request:
·HDCP Daemon required DRM Master privileges to talk with I915 Kernel driver. Hence HDCP Daemon use wayland_util to communicate with IAS (Wayland) which has DRM master privileges. Now HDCP Daemon can request enable/disable HDCP via IAS.
- Added check to differentiate whether HDCP is running with/without IAS compositor by using one of ENV.
-  Also make sure new util_wayland library have proper versioning same as other libraries

satyeshwar committed:
util_wayland: Added a Wayland library for communication with IAS

This change allows the daemon to talk with the IAS Wayland
compositor to set the content protection property.

Signed-off-by: Satyeshwar Singh <satyeshwar.singh@intel.com>

satyeshwar committed:
daemon: Changed Status to status

Status is a keyword in several header files and this
causes the daemon to not compile so just changed from upper to
lower case.

Signed-off-by: Satyeshwar Singh <satyeshwar.singh@intel.com>

saik-intel committed:
HDCP Daemon integrating with IAS Compositor

- Fixing cmake build for util_wayland which is communicate with
IAS(Wayalnd). Build util_wayland library before hdcpd which has
dependency
- Added change to make util_wayland library versioning same as with
HDCP_VERSION
- Added ENV check to identify whether hdcpd running independently or
with IAS to get DRM Master privileges accordingly

Signed-off-by: Saikishore Konda <saikishore.konda@intel.com>

saik-intel commented:
I have make sure that PR #11  is taken care , took #11 change and on made changes top of this change #11

shiqiangyu commented:
@saik-intel, We will review this patch ASAP.

shiqiangyu commented on CMakeLists.txt:
```diff
@@ -80,9 +82,13 @@ set(HDCP_VERSION
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+
```
remove these blank lines

saik-intel replied:
okay done removed as per your review

shiqiangyu commented on CMakeLists.txt:
```diff
@@ -91,12 +97,9 @@ configure_file(${CMAKE_SOURCE_DIR}/config/libhdcpsdk.pc.in
 # Set install path
 if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-    set(CMAKE_INSTALL_PREFIX "/usr/local")
+   set(CMAKE_INSTALL_PREFIX "/usr/local")
```
indention should be 4 white spaces.

shiqiangyu commented on daemon/portmanager.cpp:
```diff
@@ -1147,58 +1148,68 @@ int32_t PortManager::SetPortProperty(
 {
     HDCP_FUNCTION_ENTER;
 
+    char *dir = getenv("XDG_RUNTIME_DIR");
```
Use more meaningful variable name for IAS presence.

saik-intel replied:
updated with proper varialbe ias_env and commited change

shiqiangyu commented on daemon/portmanager.cpp:
```diff
@@ -1147,58 +1148,68 @@ int32_t PortManager::SetPortProperty(
-    {
-        HDCP_ASSERTMESSAGE("Could not drop drm master privilege");
-        return EBUSY;
+        util_set_content_protection(drmId, *value);
```
Function SetPortProperty is not only for setting content protection property. Hence this if-else branch is not appropriate.

saik-intel replied:
Whatever HDCP Daemon requesting via SetPortProperty API call , it will be directed to IAS via Util_set__content_protection. This means all requests will be directed via IAS. Since HDCP is content protection request, the util will be such name

saik-intel replied:
Currently Util_Set_Content_Protection will do for eanble/disalbe HDCP, In next checkin i will take care of SRM feature

shiqiangyu replied:
Got it, thanks.

shiqiangyu commented on daemon/util_wayland/CMakeLists.txt:
```diff
@@ -0,0 +1,34 @@
+
+include_directories( ../ )
+include_directories(${LIBDRM_INCLUDE_DIRS})
+include_directories( $ENV{WLD}/include $ENV{WLD}/include/libdrm  ../../external/libdrm ../../external/libdrm/include/drm wl_base )
```
One path one line.

saik-intel replied:
submitted proposed change in my branch  .

shiqiangyu commented on daemon/util_wayland/CMakeLists.txt:
```diff
@@ -0,0 +1,34 @@
+
+add_definitions(-g -Wall)
+
+add_library( ${PROJECT_NAME} SHARED
```
is util_wayland a open source library, if so. it's better not include all the source code into hdcp. For example, hdcp depends on libdrm, but needn't include all the libdrm source code.

satyeshwar replied:
No, util_wayland is not open source. I wrote it for another project and it fits nicely in this one so provided the source here too.

shiqiangyu replied:
@satyeshwar ok, it's fine to include these files for now, btw, is there any plan for util_wayland open source?

XiangKaren replied:
@satyeshwar, the source codes in this github is for hdcp handing, it's ok to include IPC calling to wayland, but please don't check-in util_wayland (that's Wayland specific) codes here.

saik-intel committed:
Update CMakeLists.txt

removed white lines as per review

saik-intel committed:
Update CMakeLists.txt

removing tab with 4 white spacs

saik-intel committed:
using proper varialbe ias_env instead of dir

- ias_env variable will determine whether hdcpd is running along with
IAS or not

Signed-off-by: Konda SaiKishore <saikishore.konda@intel.com>

saik-intel committed:
adding one path will be in one line for include_directories

Signed-off-by: Konda SaiKishore <saikishore.konda@intel.com>
